src/main/java/it/pagopa/interop/probing/statistics/api/util/logging/DateUtilities.java

```diff
@@ -16,10 +16,10 @@ public String changeDateFormat(String dateString) {
     return date.toOffsetDateTime().toString();
   }
 
-  public OffsetDateTime zeroDate(OffsetDateTime inputDate) {
+  public OffsetDateTime zeroDate(OffsetDateTime inputDate , boolean multipleDays) {
     DateTimeFormatter dateformatter =
         DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSS").withZone(ZoneId.systemDefault());
-    String startDateTrunc = inputDate.toString().substring(0, 13) + ":00:00.000";
+    String startDateTrunc = multipleDays ? inputDate.toString().substring(0, 11) + "00:00:00.000" : inputDate.toString().substring(0, 13) + ":00:00.000";
     ZonedDateTime date = ZonedDateTime.parse(startDateTrunc.replace("T", " "), dateformatter);
```

Comment:
Maybe it is better to use multiple ofPattern

Reply:
The function needs to reset the hours or the minutes depending on the case. In the first case i can't use a different pattern(yyyy-MM-dd) because i will obtain a date, which is not my desired type and will throw an exception on the parsing.